services/sqs/src/test/java/software/amazon/awssdk/services/sqs/SqsAsyncBatchManagerTest.java

```diff
@@ -104,15 +98,15 @@ public void tearDown() {
 
     @Test
     public void closeBatchManager_shouldNotCloseExecutorsOrClient() {
-        SqsAsyncBatchManager batchManager = new DefaultSqsAsyncBatchManager(client, scheduledExecutor,
+        SqsAsyncBatchManager batchManager = new DefaultSqsAsyncBatchManager(client,
                                                                             mockSendMessageBatchManager,
                                                                             mockDeleteMessageBatchManager,
                                                                             mockChangeVisibilityBatchManager);
         batchManager.close();
         verify(mockSendMessageBatchManager).close();
         verify(mockDeleteMessageBatchManager).close();
         verify(mockChangeVisibilityBatchManager).close();
-        assertThat(scheduledExecutor.isShutdown()).isFalse();
+        assertThatCode(() -> client.serviceName()).doesNotThrowAnyException(); // To make sure client is not closed
```

Comment:
I think if you close the client, you can still get the serviceName. Can we create a mock client and verify that client.close() is not invoked?

Reply:
Created mock client and verified c`lient.close() `is not invoked.